packages/core/src/browser/shell/tab-bars.ts

```diff
@@ -274,12 +299,21 @@ export class TabBarRenderer extends TabBar.Renderer {
     }
 
     /**
-     * Get the decoration data given the tab URI and the decoration data type.
+     * Get the decoration data given the tab URI and the decoration data type sorted by priority.
      * @param {string} title The title.
      * @param {K} key The type of the decoration data.
      */
     protected getDecorationData<K extends keyof WidgetDecoration.Data>(title: Title<Widget>, key: K): WidgetDecoration.Data[K][] {
-        return this.getDecorations(title).filter(data => data[key] !== undefined).map(data => data[key]);
+        return this.getDecorations(title)
+            .filter(data => data[key] !== undefined)
+            .sort((a, b) => this.compareDecorators(a, b))
+            .map(data => data[key]);
+    }
+
+    protected compareDecorators(a: WidgetDecoration.Data, b: WidgetDecoration.Data): number {
+        const priorityA = a.priority || 0;
+        const priorityB = b.priority || 0;
+        return priorityB - priorityA;
```

Comment:
@colin-grant-work I believe we can, but in this way we can define more easily a custom comparator specific for decorations from which downstream apps can more easily extend without impacting other areas of the code. Is this a refactoring you want as part of the changes?

Reply:
You're right about extensibilitiy, but I think it would still be slightly nicer to have one default implementation, and then have this method look like this, for now:

```typescript
compareDecorators(a, b): number {
    return SomeNamespace.DefaultComparator(a, b);
}
```

But if you feel like it isn't worth the trouble, I won't insist on it for this PR.

Reply:
@vince-fugnitto : I believe Colin would be fine wit you merging this as is :-)